### PR TITLE
Review from Tamash

### PR DESCRIPTION
**Pull request by RubelMozumder**
@tomio13 You can comment in this PR.

### CODE REVIEW

**Comment by tomio13** [comment]
In the bias scan part I have the feeling that we have multiple, overlapping time variables, we should look into.

**Review by RubelMozumder — COMMENTED** [review]
NXscan.yaml is not part of this PR. Only the base classes and app. def. that come through the SPM folder.

File: contributed_definitions/nyaml/SPM/NXcantilever_spm.yaml [hunk 1]
```diff
@@ -64,17 +66,39 @@ NXcantilever_spm(NXobject):
...
+      unit: NX_LENGTH
+    cantilever_thickness(NX_NUMBER):
+      doc: |
+        Nominal thickness of the cantilever in microns
+      unit: NX_LENGTH
+    cantilever_resonance_frequency(NX_NUMBER):
+      doc: |
+        Nominal free resonance frequency of the cantilever in air, in kHz
+      unit: NX_FREQUENCY
+
```
Comment (RubelMozumder) [inline contributed_definitions/nyaml/SPM/NXcantilever_spm.yaml:101]:
```suggestion
    cantilever_type:
      doc: |
        The shape of the cantilever as a general text, such as A-shape, beam, or arrow.

    cantilever_length(NX_NUMBER):
      doc: |
        Nominal length between base and end of the cantilever in micrometers.
      unit: NX_LENGTH
    cantilever_width(NX_NUMBER):
      doc: |
        Nominal width of the cantilever in microns.
      unit: NX_LENGTH
    cantilever_thickness(NX_NUMBER):
      doc: |
        Nominal thickness of the cantilever in microns.
      unit: NX_LENGTH
    cantilever_resonance_frequency(NX_NUMBER):
      doc: |
        Nominal free resonance frequency of the cantilever in air, in kHz.
      unit: NX_FREQUENCY
      
```

File: contributed_definitions/nyaml/SPM/NXcircuit.yaml [hunk 2]
```diff
@@ -1,12 +1,24 @@
...
-type: group
+
+  Electronic circuits are hardware components connecting several electronic components to achieve
+  specific functionality, e.g. amplifying a voltage or convert a voltage to binary numbers, etc.
+
+symbol:
+  doc: |
+    constants used in the definition: the number of channels of the circuit board
+  N_channel: |
+    number of channels of the circuit board
```
Comment (RubelMozumder) [inline contributed_definitions/nyaml/SPM/NXcircuit.yaml:12]:
```suggestion
symbol:
  doc: |
    Constant to be used in the definition: the number of channels of the circuit board.
  N_channel: |
    number of channels of the circuit board.
```

File: contributed_definitions/nyaml/SPM/NXlockin.yaml [hunk 3]
```diff
@@ -27,16 +29,21 @@ NXlockin(NXobject):
       A periodic voltage signal generated by the lock-in,
       usually applied to a sample and used to create a reference signal for the detection of the input signal
   lockin_current_flip_value(NX_NUMBER):
+  # this is an STM specific property, not used this way elsewhere
+  # maybe replacing with inverted output, which would tell whether the provided amplitude
+  # is used as is or inverted in its sign
```
Comment (RubelMozumder) [inline contributed_definitions/nyaml/SPM/NXlockin.yaml:34]:
```suggestion
  
```

File: contributed_definitions/nyaml/SPM/NXlockin.yaml [hunk 4]
```diff
@@ -48,12 +55,12 @@ NXlockin(NXobject):
...
-      Phase of the reference signal for the lock-in amplifier.
+      Phase of the reference signal set in the lock-in amplifier.
+
   demodulated_signal(NX_CHAR):
     doc: |
-      The modulated output signal will be demodulated, in order to determine the amplitude
-      and phase at the frequency set in the Frequency field or harmonics, such as current,
-      bias, et.al.
+      What it the type of the demodulated signal? current | voltage
+
```
Comment (RubelMozumder) [inline contributed_definitions/nyaml/SPM/NXlockin.yaml:63]:
```suggestion
      Type of the demodulated signal,  current | voltage.

```

File: contributed_definitions/nyaml/SPM/NXpiezo_config_spm.yaml [hunk 5]
```diff
@@ -1,9 +1,11 @@
...
 doc: |
-  A base class describing piezo settings for scanning probe microscopy.
+  A base class describing piezo actuator settings for scanning probe microscopy.
   
-  The piezoelectric material of SPM actuators gets deformed due to the applied electric field.
-  Description below shows calibration coefficients and other configuration parameters of the piezo actuators.
+  Piezoelectric actuators work utilizing the inverse-piezoelectric effect, when a voltage
+  is applied on the material and it deforms proportional to the applied voltage.
+  Description below shows calibration coefficients and other configuration parameters of open loop piezo actuators.
+  (that is actuators without capacitive sensor feedback systems)
```
Comment (RubelMozumder) [inline contributed_definitions/nyaml/SPM/NXpiezo_config_spm.yaml:8]:
```suggestion
  Description below shows calibration coefficients and other configuration parameters of open loop piezo actuators (that is actuators without capacitive sensor feedback systems).
```

**Review by RubelMozumder — COMMENTED** [review]
Comments from Tamash have been implemented.

**Pull request merged**